### PR TITLE
Laravel i18n

### LINKED ISSUE #42: Laravel
Add the capability to generate/upload/download translation files for Laravel. 

- [ ] Able to generate translation files (for php or json, since the translation strings differ https://laravel.com/docs/9.x/localization#retrieving-translation-strings)
- [ ] Able to upload to Localize correctly.
- [ ] Able to download from Localize, directly into php or json that Laravel accepts.

### PR DESCRIPTION
**Pull request by goncaloasimoes**
Handling the making of pots for laravel from php and blade files, exporting and importing via localize.

closes #42

### CODE REVIEW

File: lib/Translations/PotMaker.php [hunk 1]
```diff
@@ -46,11 +46,14 @@ public function __construct( Project $config ) {
...
-				" {$source_path}\n"
-			);
+		$sources = is_array( $source_path ) ? $source_path : [ $source_path ];
+		foreach ( $sources as $source ) {
+			if ( ! is_dir( $source ) ) {
+				throw new DirectoryDoesntExist(
+					'Source directory for generating pots does not exist. Path received was:' .
+					" {$source}\n"
+				);
+			}
```
Comment (csrui) [inline lib/Translations/PotMaker.php:56]: Removing complexity.

- 1 negation.
- 1 indentation.

```suggestion
			if ( is_dir( $source ) ) {
				continue;
			}
			throw new DirectoryDoesntExist(
				'Source directory for generating pots does not exist. Path received was:' .
				" {$source}\n"
			);
```

File: lib/Utils/CodeParser.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace TwentySixB\Translations\Utils;
+
+use KKomelin\TranslatableStringExporter\Core\CodeParser as CoreCodeParser;
+
+/**
+ * Code Parser, used for making pots in Laravel.
```
Comment (csrui) [inline lib/Utils/CodeParser.php:8]:
```suggestion
 * Code Parser, used for generating pots in Laravel.
```

File: lib/Utils/CodeParser.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+class CodeParser extends CoreCodeParser {
+	public function __construct(
+		array $functions = [ '__', '_t', '@lang' ],
+		bool $allow_newlines = false
+	) {
+		$this->functions = $functions;
+		$this->pattern   = str_replace('[FUNCTIONS]', implode('|', $this->functions), $this->pattern);
+        if ( $allow_newlines ) {
+            $this->pattern .= 's';
+        }
```
Comment (csrui) [inline lib/Utils/CodeParser.php:21]: indentation issues

File: lib/Utils/FileFinder.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace TwentySixB\Translations\Utils;
+
+use KKomelin\TranslatableStringExporter\Core\FileFinder as CoreFileFinder;
+use Symfony\Component\Finder\Finder;
+
+
+/**
+ * File finder, used for making pots in Laravel.
```
Comment (csrui) [inline lib/Utils/FileFinder.php:10]:
```suggestion
 * File finder, used for generating pots in Laravel.
```

File: lib/Utils/CodeParser.php (hunk 2 above)
Comment (csrui) [reply lib/Utils/CodeParser.php:8]: Shouldn't it be used for "extracting string from..." ?

File: lib/Clients/Generator/Laravel_Gettext.php [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+		$finder  = new FileFinder( $args['source'] );  // TODO: Handle excluded directories.
+		$parser  = new CodeParser();
+		$strings = [];
+
+        $files = $finder->find();
+        foreach ( $files as $file ) {
+            $strings = array_merge( $strings, $parser->parse( $file ) );
+        }
+
+        return array_unique( $strings );
```
Comment (csrui) [inline lib/Clients/Generator/Laravel_Gettext.php:29]: Indentation issues

File: lib/Utils/CodeParser.php (hunk 2 above)
Comment (goncaloasimoes) [reply lib/Utils/CodeParser.php:8]: They're both correct but I'll use the "extracting strings from" one 😁

**Review by csrui — APPROVED** [review]

**Comment by csrui** [comment]
@xipasduarte you have some feedback on this?

**Review by csrui — APPROVED** [review]

**Pull request merged**